src/routes/v2/authenticated/collaborators.ts

```diff
@@ -0,0 +1,135 @@
+import autoBind from "auto-bind"
+import express from "express"
+
+import { AuthorizationMiddleware } from "@middleware/authorization"
+import { attachReadRouteHandlerWrapper } from "@middleware/routeHandler"
+
+import UserWithSiteSessionData from "@classes/UserWithSiteSessionData"
+
+import { BaseIsomerError } from "@root/errors/BaseError"
+import { attachSiteHandler } from "@root/middleware"
+import { RequestHandler } from "@root/types"
+import CollaboratorsService from "@services/identity/CollaboratorsService"
+
+interface CollaboratorsRouterProps {
+  collaboratorsService: CollaboratorsService
+  authorizationMiddleware: AuthorizationMiddleware
+}
+
+// eslint-disable-next-line import/prefer-default-export
+export class CollaboratorsRouter {
+  private readonly collaboratorsService
+
+  private readonly authorizationMiddleware
+
+  constructor({
+    collaboratorsService,
+    authorizationMiddleware,
+  }: CollaboratorsRouterProps) {
+    this.collaboratorsService = collaboratorsService
+    this.authorizationMiddleware = authorizationMiddleware
+    autoBind(this)
+  }
+
+  createCollaborator: RequestHandler<
+    never,
+    unknown,
+    { email: string; acknowledge?: boolean },
+    { siteName: string },
+    { userWithSiteSessionData: UserWithSiteSessionData }
+  > = async (req, res) => {
+    const { email, acknowledge = false } = req.body
+    const { siteName } = req.params
+    const resp = await this.collaboratorsService.create(
+      siteName,
+      email,
+      acknowledge
+    )
+
+    // Check for error and throw
+    if (resp instanceof BaseIsomerError) {
+      throw resp
+    }
+    return res.sendStatus(200)
+  }
+
+  deleteCollaborator: RequestHandler<
+    never,
+    unknown,
+    never,
+    { siteName: string; userId: string },
+    { userWithSiteSessionData: UserWithSiteSessionData }
+  > = async (req, res) => {
+    const { siteName, userId } = req.params
+    const resp = await this.collaboratorsService.delete(siteName, userId)
+
+    // Check for error and throw
+    if (resp instanceof BaseIsomerError) {
+      throw resp
+    }
+    return res.sendStatus(200)
+  }
+
+  listCollaborators: RequestHandler<
+    never,
+    unknown,
+    never,
+    { siteName: string },
+    { userWithSiteSessionData: UserWithSiteSessionData }
+  > = async (req, res) => {
+    const { siteName } = req.params
+    const { userWithSiteSessionData } = res.locals
+    const collaborators = await this.collaboratorsService.list(
+      siteName,
+      userWithSiteSessionData.isomerUserId
+    )
```

Comment:
I'm wondering if we should pass the entire `userWithSiteSessionData` if possible, so that there's less code change if the userData that we store changes?

Reply:
We discussed this briefly during the notifications design doc session - my thought is that we want to be more specific with the params for the `Service`-level methods. In my mind, the `userWithSiteSessionData` object is mostly used for authentication/authorization, which is a concern handled by a separate layer.